.github/workflows/ci.yml

```
@@ -0,0 +1,31 @@
+name: Fastify Official Demo CI
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        node-version: [22]  
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v3
+      with:
+        node-version: ${{ matrix.node-version }}
+        
+    - name: Install dependencies
+      run: npm i --ignore-scripts
+      
+    - name: Lint Code
+      run: npm run lint
+      
+    - name: Test
+      run: npm test
```

```suggestion
name: CI

on:
  push:
    branches:
     - main
     - master
     - next
     - 'v*'
    paths-ignore:
      - 'docs/**'
      - '*.md'
  pull_request:
    paths-ignore:
      - 'docs/**'
      - '*.md'

jobs:
  test:
    uses: fastify/workflows/.github/workflows/plugins-ci.yml@v3
    with:
      license-check: true
      lint: true
```

Please use one of the shared workflows, it will make this easier to maintain going forward.

---

Do you agree with using ci v4?

Everything is green [here](https://github.com/jean-michelet/demo/pull/2), I used `glob` to run the tests on *windows, Node18*, maybe I should do it differently?

---

I'm setting up the database, and I think it's going to be difficult to use shared workflows at the moment. At least, I don't feel competent enough in this area to adapt the existing.

I think I can create a really easy-to-maintain CI here.
There's no need for tons of demos so I think it should be okay, wdyt?

---

Maybe I could offer a new shared workflow on the repo too, it would be easier to have all the workflows at the same place when updating versions.

---

Ok, if I think Matteo prefer I create the workflow here: https://github.com/fastify/workflows/issues/133#issuecomment-2154584252